mathics/builtin/makeboxes.py

```diff
@@ -396,7 +403,7 @@ def apply_general(self, expr, f, evaluation):
         if isinstance(expr, BoxElementMixin):
             # If we are inside a DisplayForm block,
```

Comment:
Is this really just DisplayForm that could do this? There are no other forms that are like this. Or this isn't something Form creator might want to do?

Reply:
By now, this is the only case. This could also be a way to implement `InputForm/OutputForm` and the other non-standard print forms but setting a different control symbol.  In any case, I am not sure that this way is better than the one I proposed before.

Reply:
> and the other non-standard print forms

What makes a print form "non-standard"?

Reply:
MakeBoxes accepts `StandardForm|TraditionalForm` as the second parameter. In some way, when you evaluate

```
MakeBoxes[OutputForm[expr], StandardForm]
```

Makeboxes first applies the format rules associated with `OutputForm` to  `expr` and its subexpressions. Then, the reference to `OutputForm` are lost in the expression, but in some way, MakeBoxes knows that has to apply the `OutputForm` makeboxes rules to the inner expressions. One way to simulate this behavior would be to set a state variable that remembers that we are using OutputForm rules. This is not the way in which I think we should implement this, but it is a possibility.

Reply:
> but in some way, MakeBoxes knows that has to apply the OutputForm makeboxes rules to the inner expressions. 

https://reference.wolfram.com/language/ref/TraditionalForm.html says:

> TraditionalForm inserts invisible TagBox and InterpretationBox constructs into the box form of output it generates, to allow unique interpretation.tionalForm

Does this provide a mechanism for how to do this. If so, possibly the same is could done for StandardForm (and is done but just not mentioned). Sometimes it happens that only in implementing the second thing of a kind the general mechanism, and the first is done in a more hacky way is not retrofitted.